proxy-node-translator/src/test/java/uk/gov/ida/notification/softhsm/SoftHSMIntegrationTest.java

```diff
@@ -0,0 +1,171 @@
+    private static final String CRED_ENV = "dev";
+    private static final String CRED_PIN = "1234";
+    private static final String CRED_SLOT = "0";
+    private static final String CRED_NAME = "vfpn-uk";
+    private static final String CRED_ALIAS = CRED_ALIAS_PREFIX + "-" + CRED_ENV;
+
+    // softHSM Configuration
+    private static String softHSMLibPath = System.getenv("SOFT_HSM_LIB_PATH");
+
+    /**
+     * softHSM tests should only run if softHSM is installed and configured.
+     * Installation and configuration to be indicated by the SOFT_HSM_LIB_PATH environment variable.
+     * softHSM is installed in the Translator Docker image and this variable is set, so the tests
+     * will run on pre-commit.  Running the tests when building locally is effectively excluded
+     * by not setting this environment variable.
+     */
+    @Before
+    public void checkSoftHSMConfigured() {
+        assumeTrue(softHSMLibPath != null && softHSMLibPath.length() > 0);
+    }
+
+    @Test(expected = java.lang.NullPointerException.class)
+    public void shouldThrowExceptionForPKCS11ProviderConfigurationWithoutCredentials() throws Exception {
+
+        PKCS11ProviderConfiguration configuration = createPKCS11SoftHSMProviderConfigurationWithoutCredentials(CRED_SLOT, CRED_NAME);
+        PKCS11Provider pkcs11Provider = getPkcs11ProviderForConfiguration(configuration);
+
+        TestKeyPair signingTestKeyPair = new TestKeyPair(PROXY_NODE_SIGNING_CRT, PROXY_NODE_SIGNING_PK_8);
+        Credential credential = new PKCS11Credential(
+                signingTestKeyPair.certificate,
+                pkcs11Provider.getProviderNameList(),
+                CRED_ALIAS_PREFIX, CRED_PIN);
+    }
+
+    @Test
+    public void shouldCreateCredentialConfigurationList() {
+        PKCS11ProviderConfiguration configuration = createPKCS11SoftHSMProviderConfigurationWithCredentials(CRED_ALIAS, CRED_ENV, CRED_PIN, CRED_SLOT, CRED_NAME);
+        assertNotNull("PKCS11ProviderConfiguration not loaded from credentials", configuration);
+    }
+
+
+    @Test
+    public void shouldCreatePKSC11ProviderFromFactory() throws Exception {
+
+        PKCS11ProviderConfiguration configuration = createPKCS11SoftHSMProviderConfigurationWithCredentials(CRED_ALIAS, CRED_ENV, CRED_PIN, CRED_SLOT, CRED_NAME);
+        PKCS11Provider pkcs11Provider = getPkcs11ProviderForConfiguration(configuration);
+        assertNotNull(pkcs11Provider);
+    }
+
+
+    @Test
+    public void shouldRetrieveCredentialsFromHSM() throws Exception {
+
+        PKCS11ProviderConfiguration configuration = createPKCS11SoftHSMProviderConfigurationWithCredentials(CRED_ALIAS, CRED_ENV, CRED_PIN, "162624334", CRED_NAME);
+        PKCS11Provider pkcs11Provider = getPkcs11ProviderForConfiguration(configuration);
+        TestKeyPair signingTestKeyPair = new TestKeyPair(PROXY_NODE_SIGNING_CRT, PROXY_NODE_SIGNING_PK_8);
+
+        addPrivateKeyToKeyStore(signingTestKeyPair, CRED_ALIAS, CRED_PIN);
```

Comment:
What's happening here? `addPrivateKeyToKeyStore ` doesn't seem to return anything - does it  mutate the `signingTestKeyPair` in some way?